src/auth.py

```diff
@@ -0,0 +1,50 @@
+import requests
+from fastapi import Depends, HTTPException, Request
+from jwcrypto import jwk, jwt
+
+from src.config import settings
+
+# The URL to the JWKS endpoint
+JWKS_URL = settings.JWKS_URL
+
+def get_jwk():
+    if settings.JWKS_CACHE:
+        key = jwk.json_decode(settings.JWKS_CACHE).get("keys")[0]
+        return jwk.JWK(**key) 
+
+    # Make a request to get the JWKS
+    jwks = ""
+    try:
+        jwks = requests.get(JWKS_URL).text
```

Comment:
suggest `jwks = jwk.JWKSet.from_json(response.text)` and just returning that instead of a single key. you can also pass this in directly into the `key` parameter when decoding the JWT!